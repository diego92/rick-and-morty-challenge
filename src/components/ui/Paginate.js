import Pagination from "@mui/material/Pagination";
import Stack from "@mui/material/Stack";
import { Grid } from "@mui/material";

export default function Paginate({ count, handleChangePage }) {
  const handleChangePaginate = (e, page) => {
    handleChangePage(page);
  };

  return (
    <Grid
      container
      direction="row"
      justifyContent="center"
      alignItems="center"
      style={{ margin: "1rem" }}
    >
      <Stack spacing={2}>
        <Pagination
          count={count}
          color="primary"
          onChange={handleChangePaginate}
        />
      </Stack>
    </Grid>
  );
}
