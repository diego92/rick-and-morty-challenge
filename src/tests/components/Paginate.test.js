import { shallow } from "enzyme";
import Paginate from "../../components/ui/Paginate";

describe("Pruebas en <Paginate/>", () => {
  test("debe mostrarse correctamente", () => {
    const wrapper = shallow(<Paginate count={10} pageNumber={3} />);

    expect(wrapper).toMatchSnapshot();
  });
});
