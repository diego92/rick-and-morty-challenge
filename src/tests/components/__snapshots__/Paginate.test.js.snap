// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Paginate/> debe mostrarse correctamente 1`] = `
<ForwardRef(Grid)
  alignItems="center"
  container={true}
  direction="row"
  justifyContent="center"
  style={
    Object {
      "margin": "1rem",
    }
  }
>
  <ForwardRef(Stack)
    spacing={2}
  >
    <ForwardRef(Pagination)
      color="primary"
      count={10}
      onChange={[Function]}
      page={3}
    />
  </ForwardRef(Stack)>
</ForwardRef(Grid)>
`;
